aas_core_codegen/java/xmlization/_generate.py

```diff
@@ -613,24 +609,26 @@ def _generate_deserialize_cls_property(prop: intermediate.Property) -> Stripped:
 
     xml_prop_name_literal = java_common.string_literal(naming.xml_property(prop.name))
 
+    cls_name = java_naming.class_name(cls.name)
+
     return Stripped(
         f"""\
-Result<{target_cls_name}> {try_target_var} = {target_cls_name}FromSequence(
+Result<{target_cls_name}> {try_target_var} = try{target_cls_name}FromSequence(
 {I}reader, isEmptyProperty);
 
 if ({try_target_var}.isError()) {{
 {I}{try_target_var}.getError()
 {II}.prependSegment(
 {III}new Reporting.NameSegment(
 {IIII}{xml_prop_name_literal}));
-{I}return Result.failure({try_target_var});
+{I}return {try_target_var}.castTo({cls_name}.class);
```

Comment:
Is the cls_name (and thus the cls argument) necessary here? Isn't this simply target_cls_name?

Reply:
Here we need to use cls_name for the cast. target_cls_name is the class name for the property here.